_posts/2018-12-16-simple-front-end.md

```diff
@@ -0,0 +1,96 @@
+
+### GLAMP
+The Good old LinuxApacheMysqlPhp stack. This isn't strictly LAMP, but generally anything which renders dynamic data into HTML on the back end and returns it to the browser.
+
+* Synchronous
+	* What's shown to the user is driven by a user interaction resulting in loading new pages 
+* Server side rendering
+#### Pros
+* Can be very performant on low end devices since there is little or no javascript running
+* Pages with data from many sources may load faster because of better network performance between the front end server and the data sources compared to the browser and the back end / data sources
+#### Cons
+* Performance scales poorly, because pages cannot be cached
+* New content requires a page reload
+
+### SPA
+The modern Single Page Application is largely script driven. 
+
+* Asynchronous 
+	* What's shown on to the user is driven by user interaction or programmatic event resulting in scripts changing some content on the current page
+* Client side rendering
+	* Little or no HTML is delivered to the browser. Rather, javscript creates all the elements on the page.
+
+#### Pros
+* Most frameworks provide powerful two-way data binding to template and update what's on the page at any time without reloading
+* Feels more like a traditional desktop application 
+* Static scripts can be hosted on a CDN and scale well
+#### Cons
+* May perform poorly on lower end devices due to being script driven
+* First load may be slow due to large script dependencies and entire app needing to be sent at once
+* Mix of versioning, technologies, and compatibility can result in complicated build processes with polyfilling, transpiling, and dependency resolution
+
+### Somewhere in the middle
+What we'll cover today.
+* (mostly) Synchronous
+* Client side rendering
+	* Static HTML is delivered to the browser, but a small script provides the data for the dynamic content
+#### Pros
+* Performs well across a range of devices
+* Static parts of pages can be cached and hosted on a CDN
+#### Cons
+* New or refreshed content is accessed by a page reload, but that shouldn't be as much of a problem as it is with server side rendering since most of the page is cached
+
+## Solution
+
+So what are we actually talking about? In short it's two (three) dependencies:
+### Bootstrap
+Bootstrap makes it easy to quickly set up the style and layout of a page. It's been around since 2011 and hasn't changed too radically since then, but it has grown an abundance of documentation, community support, and usage guides. If you were inclined you could swap this out with any other library that does styling and layout like Google's material design for web. 
+### jQuery 
+jQuery is a transitive dependency from bootstrap, we'll only really use it directly to DRY up parts of our pages like headers and footers. When it comes to web libraries that are widely used, stable, and not going anywhere there is nothing that comes close to jQuery.
```

Comment:
I saw how you did with the headers and footers. A few comments here:
- Particularly I wouldn't import jQuery to load headers and footers. Use partials for that.
- As far as the jQuery version, you want to use `jquery-3.3.1.slim.min.js` instead of `jquery-3.3.1.min.js`.